### PR TITLE
Add support for HPX 1.7.1

### PR DESCRIPTION
**Pull request by G-071**
This PR adds support for compiling Octo-Tiger with HPX 1.7.1

Changelog:
- Updated HPX headers
- Splitted options.cpp into a separate library (to avoid cyclical build dependencies)
- Added more device guards for Kokkos compilation as required

Build instructions
- The required changes in the toolchain can be found here: STEllAR-GROUP/OctoTigerBuildChain#12

Caveats:
- Using newer HPX versions requires us to use updated Kokkos and hpx-kokkos versions. Hence, this PR currently still depends on the Kokkos PR containing the updates to the recent hpx API changes.
- ~~In its current form, this PR would raise the minimum required HPX version to 1.7.0. However, it should be possible to support older versions with some additional changes regarding the headers and buildscripts.~~

Remaining ToDos:
- [x] Support older HPX versions?
- ~~Update CI tests~~ Keep Jenkins Scripts on 1.6.0 for now until we actually use 1.7.1 in production

### CODE REVIEW

**Comment by hkaiser** [comment]
Thanks Gregor for looking into this!

File: frontend/frontend-helper.cpp [hunk 1]
```diff
@@ -42,7 +42,11 @@
 #include <hpx/include/actions.hpp>
 #include <hpx/include/lcos.hpp>
 #include <hpx/include/util.hpp>
-#include <hpx/modules/runtime_distributed.hpp>  
+
+// header not needed for HPX versions <= 1.6
+#if HPX_VERSION_MAJOR > 1 || (HPX_VERSION_MAJOR == 1 && HPX_VERSION_MINOR > 6)
```
Comment (hkaiser) [inline frontend/frontend-helper.cpp:47]:
```suggestion
#if HPX_VERSION_FULL > 0x010106
```

Comment (hkaiser) [reply frontend/frontend-helper.cpp:47]: What functionality did you miss without that header file?

**Comment by diehlpk** [comment]
@G-071 I could build Octo-Tiger using Kokkos with HPX 1.7.1 and will do some testing soon.

**Comment by diehlpk** [comment]
@G-071 I think we can merge that pull request. I could run on Perlmutter using HPX 1.7.1.

However, for Kokkos there are some build issues and I will open a new ticket.

Same for the crashing if Kokkos kernels are enabled. This seems not to related to this pull request.

**Review by shibersag — COMMENTED** [review]
@G-071, Can you explain why this change in options.cpp is required?
Is it just a name change or?

**Comment by diehlpk** [comment]
@G-071 Can we merge that to master? I would prefer to move to HPX 1.7.1 because I debug the MPI issues with this branch?

**Comment by G-071** [comment]
@diehlpk From my point of view it's ready to be merged, but I'm not the one reviewing it :-)

@shibersag Do you have any more questions about this PR?

**Comment by diehlpk** [comment]
@hkaiser Could you please have a look?I really would like to get that to master soon.

**Review by diehlpk — APPROVED** [review]
LGTM! I tested Octo-Tiger on Perlmutter with this branch and everything worked.

**Pull request merged**